Filters/Average.cs

```diff
@@ -3,48 +3,61 @@
+    /// <summary> 
+    /// The Average filter is intended for use as an average-gathering system, using a rolling average 
+    /// with "roll-length" <c>FilterCount</c> or continuous average.
+    /// </summary>
     /// <remarks>
-    /// Implementation Details:
-    /// 
-    /// *Construct Average filter given a rolling
-    ///  filter length, <c>FilterCount</c>
-    /// 
-    /// *Iteratively add values into the filter
-    ///  using <c>Feed(T Input)</c>
-    /// 
-    /// *Get the filter output by calling
-    ///  <c>YourFilterInstance.Output</c>
+    /// Intended Usage:
+    /// * Construct Average filter given a rolling
+    ///   filter length, <see cref="FilterCount">FilterCount</see>. Setting
+    ///   FilterCount to null puts the filter into
+    ///   continuous mode.
+    /// * Iteratively add values into the filter
+    ///   using <see cref="Feed(T)">Feed()</see>
+    /// * Get the filter output by calling
+    ///   <see cref="GetOutput">YourFilterInstance.GetOutput()</see>
     /// </remarks>
     /// <typeparam name="T"> A type, which must be a numeric. </typeparam>
     public class Average<T> : IFilter<T> where T : IComparable
     {
-
+        private readonly int? FilterCount;
         private T Output; // Filter output
         private dynamic[] AverageArray; // Stored average array
-        private dynamic CurSum;         // Current sum of the average array 
-        private int FilterCount,        // Size of the average array
-                    Index,              // Current index of the average array
-                    Iterations;         // Number of iterations in the filter
+        private dynamic CurSum; // Current sum of the average array 
+        private int Index, // The current index of the average filter (when in roll mode)
+                    Iterations; // The number of iterations through the average filter
         private int NumCyclesAverageSame; // Number of cycles that the average has stayed the same
 
-        /// <summary> Construct an average filter with given roll-length. </summary>
-        /// <param name="FilterCount"> Roll length for the average filter. </param>
-        public Average(int FilterCount = 10)
+        /// <summary> Construct an average filter with given roll-length; give null for continuous mode.</summary>
+        /// <remarks> To use this as a continuous (non-rolling) average filter, set <c>FilterCount</c> to null </remarks>
+        /// <param name="FilterCount"> 
+        /// Roll length for the average filter. 
+        /// If this value is null, the filter is set to continuous mode (as opposed to a roll mode). 
+        /// </param>
+        public Average(int? FilterCount = 10)
         {
+            // Assert that T is a numeric type
             if (!UtilData.IsNumericType(typeof(T)))
             {
                 Log.Output(Log.Severity.ERROR, Log.Source.OTHER, "Average filter cannot be instantiated with non-numeric type.");
```

Comment:
I don't think both are necessary here. The exception has the text needed, so log output is redundant.